### PR TITLE
[TS] Updated the gas fee payer example to adopt the new API

### PR DESCRIPTION
**Pull request by junkil-park**
### Description

* Updated the gas fee payer example to adopt the new API of aptos SDK v. 1.17.0
* Simplifies the flow significantly

### Test Plan
`pnpm run gas_fee_payer`

### CODE REVIEW

**Review by 0xmaayan — APPROVED** [review]
LGTM

File: ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 } from "aptos";
 import { NODE_URL, FAUCET_URL } from "./common";
 import { assert } from "console";
+import * as Gen from "../../src/generated";
```
Comment (junkil-park) [inline ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts:19]: @0xmaayan , is there a better way to import this than using a relative path like this?

**Comment by xbtmatt** [comment]
Great! A lot simpler :D

Would be nice to have alice and bob's balance tracked from before and after and then print out the gas fees so you can see that Alice's balance didn't change and Bob's did, in the same amount as the gas fees paid:

```typescript
  const aliceBalanceBefore = await getBalance(alice);
  const bobBalanceBefore = await getBalance(bob);
  console.log(`Alice's balance: ${aliceBalanceBefore} octas`);
  console.log(`Bob's balance: ${bobBalanceBefore} octas`);
  
  // ... submit tx
  
  const response = await client.waitForTransactionWithResult(txnHash, { checkSuccess: true });
  const { gas_used, gas_unit_price } = response as any;
  
  // ... print token amt
  
  // Check that Alice did not pay the fee, but Bob did.
  const aliceBalanceAfter = await getBalance(alice);
  const bobBalanceAfter = await getBalance(bob);
  console.log(`Alice's balance: ${aliceBalanceAfter} octas`);
  console.log(`Bob's balance: ${bobBalanceAfter} octas`);

  console.log(
    "\n=== Check Bob and Alice's balance changes ===\n" +
      `Alice's balance change: ${aliceBalanceBefore - aliceBalanceAfter} octas\n` +
      `Bob's balance change: ${bobBalanceBefore - bobBalanceAfter} octas\n` +
      `The gas fees were: ${gas_used * gas_unit_price} octas`,
  );
```

**Review by xbtmatt — APPROVED** [review]

File: ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts (hunk 1 above)
Comment (0xmaayan) [reply ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts:19]: You can import 'Type' and then use it 'Type.<thetypeyouneed>'

Comment (junkil-park) [reply ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts]: Applied. Thank you!

**Comment by junkil-park** [comment]
> Great! A lot simpler :D
> 
> Would be nice to have alice and bob's balance tracked from before and after and then print out the gas fees so you can see that Alice's balance didn't change and Bob's did, in the same amount as the gas fees paid:
> 
> ```ts
>   const aliceBalanceBefore = await getBalance(alice);
>   const bobBalanceBefore = await getBalance(bob);
>   console.log(`Alice's balance: ${aliceBalanceBefore} octas`);
>   console.log(`Bob's balance: ${bobBalanceBefore} octas`);
>   
>   // ... submit tx
>   
>   const response = await client.waitForTransactionWithResult(txnHash, { checkSuccess: true });
>   const { gas_used, gas_unit_price } = response as any;
>   
>   // ... print token amt
>   
>   // Check that Alice did not pay the fee, but Bob did.
>   const aliceBalanceAfter = await getBalance(alice);
>   const bobBalanceAfter = await getBalance(bob);
>   console.log(`Alice's balance: ${aliceBalanceAfter} octas`);
>   console.log(`Bob's balance: ${bobBalanceAfter} octas`);
> 
>   console.log(
>     "\n=== Check Bob and Alice's balance changes ===\n" +
>       `Alice's balance change: ${aliceBalanceBefore - aliceBalanceAfter} octas\n` +
>       `Bob's balance change: ${bobBalanceBefore - bobBalanceAfter} octas\n` +
>       `The gas fees were: ${gas_used * gas_unit_price} octas`,
>   );
> ```

Nice suggestion! I updated it to display the gas fee used per transaction!

**Pull request merged**